### PR TITLE
implement cli-configurable global gas cap for RPC calls

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- This PR is derived from https://github.com/ethereum/go-ethereum/pull/19401
- This is to avoid attacks from strangers by long-running calls

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yoomee1313** [comment]
Is it okay not to check `globalGasCap` at `DoCall` function?

**Comment by ehnuje** [comment]
> Is it okay not to check `globalGasCap` at `DoCall` function?

Thanks for your checking! My mistake. I've updated the change.

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -1490,6 +1495,8 @@ func SetKlayConfig(ctx *cli.Context, stack *node.Node, cfg *cn.Config) {
 	cfg.RestartTimeOutFlag = ctx.GlobalDuration(RestartTimeOutFlag.Name)
 	cfg.DaemonPathFlag = ctx.GlobalString(DaemonPathFlag.Name)
 
+	cfg.RPCGasCap = new(big.Int).SetUint64(ctx.GlobalUint64(RPCGlobalGasCap.Name))
```
Comment (KimKyungup) [inline cmd/utils/flags.go:1498]: We need to check the flag is set or not.
Because, default is now gas cap. In that case, `cfg.RPCGasCap` should be nil.

Comment (ehnuje) [reply cmd/utils/flags.go:1498]: As it is checked by `hi > gasCap.Uint64()`, I think 0 value can be used without checking. 
Please correct me if I am wrong 😅

Comment (aidan-kwon) [reply cmd/utils/flags.go:1498]: I agree with @KimKyungup. When the flag is not set, `cfg.RPCGasCap` and `hi` will be set to `0` also.

Comment (KimKyungup) [reply cmd/utils/flags.go:1498]: Using default `0` value is not right for `doCall` and `esitmateGas`. 
In geth PR, there is a routine to check. We shouldn't skip it.

Comment (ehnuje) [reply cmd/utils/flags.go]: Understood. Updated!

**Review by KimKyungup — APPROVED** [review]

**Comment by ehnuje** [comment]
@yoomee1313 @aidan-kwon @winnie-byun @jeongkyun-oh 
Dear reviwers, could you please review this change?

**Review by yoomee1313 — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Pull request merged**